server/bin/www

```diff
@@ -21,14 +21,41 @@ app.set("port", port);
  * Create HTTP server.
  */
 
+/**
+ * user Functions
+ */
+const { addUser, getUser } = require("../users.js");
+
 var server = http.createServer(app);
 /**
  * socket io connection
  */
 const io = require("socket.io")(server);
-io.on("connection", (socket) => {
+io.on("connection", socket => {
   console.log("user conected");
 
+  socket.on("join", ({ data }) => {
+    const userId = data.userId;
+    const room = data.chatId;
+    addUser({ userId, room });
+    socket.join(room);
+  });
+
+  socket.on("message", (messageData, callback) => {
+    const chatId = messageData.chatId;
+    const userId = messageData.sender;
+    const user = getUser({ userId, chatId });
```

Comment:
I'm not sure about this approach of making each chat a room and swapping the user between them. I assume this function exists because if I'm viewing my chat with one person and a second messages me then I want that update. In that case would it be easier to just make a room with the user id and emit the message to them? That could be an approach for @Tomology 's issue as well

Reply:
So in this case, if I understand right, we have each user sitting in separate room. We emit message directly to userId (recepient), but on sender's chat we would never know if message passed to socket succesfully and will just update chat on FE, assuming it went well. Or we will need to have a second emit (back to sender, basically) to update their chat?

Reply:
If the POST for the message is successful then the message will (eventually) be received even if the socket fails, so maybe you can update the frontend if the POST is successful and then you only need to emit to the recipient